### PR TITLE
Fix hover hit on vertical and horizontal segments

### LINKED ISSUE #9458: [BUG] HoverTool in 'vline' mode doesn't work with vertical segments
Bokeh version: 1.4
Python version: 3.7.5
Browser: Firefox / Chromium
OS: Linux

When adding a `HoverTool` with `mode='vline'` to a plot with vertical segments, the tooltip is not displayed.

Example code:

```
from bokeh.io import output_file, save
from bokeh.models import HoverTool
from bokeh.plotting import figure

x = [1, 2, 3, 4]
y0 = [1] * 4
y1 = [2] * 4
fig = figure(tools=[HoverTool(mode='vline')])
fig.segment(x, y0, x, y1)
output_file('vline.html')
save(fig)
```

Screenshot:
image

### PR DESCRIPTION
**Pull request by kumaran-14**
- [x] issues: fixes #9458
- [ ] tests added / passed
- [ ] release document entry (if new feature or API change)

### CODE REVIEW

File: bokehjs/src/lib/models/glyphs/segment.ts [hunk 1]
```diff
@@ -116,6 +116,18 @@ export class SegmentView extends GlyphView {
...
       if ((v0[i] <= val && val <= v1[i]) || (v1[i] <= val && val <= v0[i]))
         hits.push(i)
+
+      if (v0[i] == v1[i]) {
+        if (geometry.direction == 'h') {
+          if (Math.abs(this.sx0[i] - sx) <= 2.5) {
+            hits.push(i)
+          }
+        } else {
+          if (Math.abs(this.sy0[i] - sy) <= 2.5) {
```
Comment (mattpap) [inline bokehjs/src/lib/models/glyphs/segment.ts]: Let's fold this magic constant into `const threshold = 2.5` with a comment explaining the choice of such constant.

File: bokehjs/src/lib/models/glyphs/segment.ts [hunk 2]
```diff
@@ -116,6 +116,20 @@ export class SegmentView extends GlyphView {
     for (const i of candidates) {
       if ((v0[i] <= val && val <= v1[i]) || (v1[i] <= val && val <= v0[i]))
         hits.push(i)
+
+      const threshold = 2.5 // Maximum pixel difference to detect hit
```
Comment (bryevdv) [inline bokehjs/src/lib/models/glyphs/segment.ts]: We should probably try computing this value as the `line_width/2` plus some padding(maybe ~1.5?) that way this should work with segments that are "thick" (i.e. more than 1 pixel wide)

**Comment by bryevdv** [comment]
@kumaran-14 do you hav a short test script that you have been using to test with that you can post here?

**Comment by kumaran-14** [comment]
```
from bokeh.io import output_file, save
from bokeh.models import HoverTool
from bokeh.plotting import figure

x = [2, 2.2, 2.2, 2.6]
x0 = [2, 2.6, 2.6, 2.8]
y0 = [1.5, 1.95, 1.2, 1.6]
y1 = [1.75, 1.95, 1.6, 1.2]
fig = figure(tools=[HoverTool(mode='vline')])
fig.segment(x, y0, x0, y1)
output_file('vline.html')
save(fig)
```

**Comment by bryevdv** [comment]
Tested vline/hline locally, thantks @kumaran-14

**Pull request merged**